shared/enclave/server/views/pin.ts

```diff
@@ -0,0 +1,246 @@
+        border: none;
+        padding: 0;
+        cursor: pointer;
+        color: #FFF;
+        outline: none;
+      }
+    </style>
+  </head>
+  <body>
+    <h1>Enter your PIN</h1>
+    <h2>Look at your device for the number positions</h2>
+
+    <div class="pin-buttons">
+      <button class="pin-button" data-number="7"></button>
+      <button class="pin-button" data-number="8"></button>
+      <button class="pin-button" data-number="9"></button>
+
+      <button class="pin-button" data-number="4"></button>
+      <button class="pin-button" data-number="5"></button>
+      <button class="pin-button" data-number="6"></button>
+
+      <button class="pin-button" data-number="1"></button>
+      <button class="pin-button" data-number="2"></button>
+      <button class="pin-button" data-number="3"></button>
+    </div>
+
+    <div class="pin-controls">
+      <input class="pin-input" type="password" readonly/>
+      <button class="pin-clear">
+        <svg viewPort="0 0 8 8" width="8" height="8" version="1.1" xmlns="http://www.w3.org/2000/svg">
+          <line x1="1" y1="7" x2="7" y2="1" stroke="white" stroke-width="2"/>
+          <line x1="1" y1="1" x2="7" y2="7" stroke="white" stroke-width="2"/>
+        </svg>
+      </button>
+
+      <button class="pin-unlock">
+        Unlock
+      </button>
+    </div>
+
+    <button class="close">
+      <svg viewPort="0 0 12 12" width="12" height="12" version="1.1" xmlns="http://www.w3.org/2000/svg">
+        <line x1="1" y1="11" x2="11" y2="1" stroke="white" stroke-width="2"/>
+        <line x1="1" y1="1" x2="11" y2="11" stroke="white" stroke-width="2"/>
+      </svg>
+    </button>
+
+    <script nonce="${scriptNonce}">
+      var ipcRenderer = require('electron').ipcRenderer;
+      var remote = require('electron').remote;
+      var pin = [];
+      var pinButtons = document.querySelectorAll('.pin-button');
+      var input = document.querySelector('.pin-input');
+      var unlock = document.querySelector('.pin-unlock');
+      var clear = document.querySelector('.pin-clear');
+      var close = document.querySelector('.close');
+
+      var closeWindow = function() {
+        var window = remote.getCurrentWindow();
+        window.close();
+      }
+
+      pinButtons.forEach(function(el) {
+        el.addEventListener('click', function(ev) {
+          pin.push(ev.currentTarget.getAttribute('data-number'));
+          input.value = pin.join('');
+        });
+      });
+
+      clear.addEventListener('click', function() {
+        pin = [];
+        input.value = '';
+      });
+
+      unlock.addEventListener('click', function() {
+        if (!pin.length) return;
+        ipcRenderer.send('${EVENT}', pin.join(''));
+        closeWindow();
+      });
+
+      close.addEventListener('click', function() {
+        closeWindow();
+      });
+    </script>
+  </body>
+</html>
+`;
+
+export async function showPinPrompt(): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
```

Comment:
Shouldnt need to specify the generic here either, can keep it in the function return statement (or even leave that empty too and let TS infer it)

Reply:
I thought so too, but it was barking at me that it was implicitly any so I just typed it to stop it. Not sure why it wasn't inferring it here.

Reply:
Nvm removing it gave me no errors. Thanks!